ui/ducks/ens.js

```diff
@@ -0,0 +1,183 @@
+import { createSlice } from '@reduxjs/toolkit';
+import ENS from 'ethjs-ens';
+import log from 'loglevel';
+import networkMap from 'ethereum-ens-network-map';
+import { isConfusing } from 'unicode-confusables';
+import { isHexString } from 'ethereumjs-util';
+
+import { getCurrentChainId } from '../selectors';
+import {
+  CHAIN_ID_TO_NETWORK_ID_MAP,
+  MAINNET_NETWORK_ID,
+} from '../../shared/constants/network';
+import {
+  CONFUSING_ENS_ERROR,
+  ENS_ILLEGAL_CHARACTER,
+  ENS_NOT_FOUND_ON_NETWORK,
+  ENS_NOT_SUPPORTED_ON_NETWORK,
+  ENS_NO_ADDRESS_FOR_NAME,
+  ENS_REGISTRATION_ERROR,
+  ENS_UNKNOWN_ERROR,
+} from '../pages/send/send.constants';
+import { isValidDomainName } from '../helpers/utils/util';
+import { CHAIN_CHANGED } from '../store/actionConstants';
+import {
+  BURN_ADDRESS,
+  isBurnAddress,
+  isValidHexAddress,
+} from '../../shared/modules/hexstring-utils';
+
+// Local Constants
+const ZERO_X_ERROR_ADDRESS = '0x';
+
+const initialState = {
+  stage: 'UNINITIALIZED',
+  resolution: null,
+  error: null,
+  warning: null,
+  network: null,
+};
+
+export const ensInitialState = initialState;
+
+const name = 'ENS';
+
+let ens = null;
+
+const slice = createSlice({
+  name,
+  initialState,
+  reducers: {
+    processEnsError: (state, action) => {
+      const { recipient, network, reason } = action.payload;
+      if (
+        isValidDomainName(recipient) &&
+        reason.message === 'ENS name not defined.'
+      ) {
+        state.error =
+          network === MAINNET_NETWORK_ID
+            ? ENS_NO_ADDRESS_FOR_NAME
+            : ENS_NOT_FOUND_ON_NETWORK;
+      } else if (reason.message === 'Illegal Character for ENS.') {
+        state.error = ENS_ILLEGAL_CHARACTER;
+      } else {
+        log.error(reason);
+        state.error = ENS_UNKNOWN_ERROR;
+      }
+    },
+    processEnsResult: (state, action) => {
+      state.resolution = action.payload;
```

Comment:
In the case of an error, I think this should remain unset. At least I think that's how it worked previously.

Reply:
good catch. Fixed here https://github.com/MetaMask/metamask-extension/pull/10965/commits/251ba67938f7758fd3d8ded3e27262d022c05a23